src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -274,7 +257,23 @@ internal string CreateFileFromContents(int runId, string fileName)
         // Contents are embedded in SARIF. Create a file from these contents.
         internal string CreateFileFromContents(IDictionary<string, ArtifactDetailsModel> fileDetailsDictionary, string fileName)
         {
+            if (!fileDetailsDictionary.ContainsKey(fileName))
```

Comment:
>ContainsKey [](start = 39, length = 11)

Check  if you have TryGetValue.
IF yes, you could change to:

if (!fileDetailsDctionary.TryGetValue(filename, out  ArtifactDetailsModel fileData)
{
  return  null;
}

Reply:
done

---
In reply to: [601282161](https://github.com/microsoft/sarif-visualstudio-extension/pull/384#discussion_r601282161) [](ancestors = 601282161)